webapp/commonComponents/validationFieldMessages.js

```diff
@@ -0,0 +1,112 @@
+
+const getMessages = (fn, severity) => i18n => R.pipe(
+  fn,
+  R.map(getErrorText(i18n)),
+  R.join(', '),
+  msg => msg ? [severity, msg] : []
+)
+
+const getValidationErrorMessages = i18n => R.converge(
+  R.concat, [
+    getMessages(Validation.getWarnings, ValidationResult.severity.warning)(i18n),
+    getMessages(Validation.getErrors, ValidationResult.severity.error)(i18n)
+  ]
+)
+
+const getValidationFieldErrorMessage = (i18n, field) => R.pipe(
+  getValidationErrorMessages(i18n),
+  R.ifElse(
+    R.isEmpty,
+    () => getErrorText(i18n)(
+      ValidationResult.newInstance(
+        Validation.messageKeys.invalidField, //default error message
+        { field }
+      )
+    ),
+    R.identity
+  )
+)
+
+const getChildrenCountValidation = (validation, survey, showKeys, i18n) => {
+  return Object.entries(validation).map(([nodeDefUuid, nodeValidation]) => {
+    const [field, fieldValidation] = Object.entries(Validation.getFieldValidations(nodeValidation))[0]
+    const nodeDef = Survey.getNodeDefByUuid(nodeDefUuid)(survey)
+    fieldValidation.errors[0].params.nodeDefName = NodeDef.getLabel(nodeDef, i18n.lang)
+    const [severity, message] = getValidationFieldErrorMessage(i18n, field)(fieldValidation)
+
+    return ([severity, `${showKeys ? `${i18n.t(field)}: ` : ''}${message}`])
+  })
+}
+
+const getValidationFieldMessages = (i18n, showKeys = true, survey) => validation => {
+  const messages = []
+
+  R.pipe(
+    Validation.getFieldValidations,
+    Object.entries,
+    // extract invalid fields error messages
+    R.forEach(([field, childValidation]) => {
+      // if it's a childrenCount validation, extract the actual minCount or maxCount validation
+      // and add the nodeDefUuid name as param
+      if (field === RecordValidations.keys.childrenCount) {
+        const childrenCountValidation = getChildrenCountValidation(Validation.getFieldValidations(childValidation), survey, showKeys, i18n)
+        messages.push(...childrenCountValidation)
+      } else {
+        const [severity, message] = getValidationFieldErrorMessage(i18n, field)(childValidation)
+        messages.push([severity, `${showKeys ? `${i18n.t(field)}: ` : ''}${message}`])
+      }
+    }),
+  )(validation)
+
+  R.pipe(
+    getValidationErrorMessages(i18n),
+    x => { if (x.length) messages.push(x) }
+  )(validation)
+
+  return messages
+}
+
+const ValidationFieldMessages = ({ validation, showKeys = true, showIcons = false, survey }) => {
+  const i18n = useI18n()
+
+  return R.pipe(
+    getValidationFieldMessages(i18n, showKeys, survey),
+    R.addIndex(R.map)(([type, message], i) =>
+      <div className={`validation-field_message ${type}`} key={i}>
+        { showIcons && <span className="icon icon-warning icon-12px icon-left" /> }
+        <Markdown className="validation-field-message__text" source={message}/>
+      </div>
+    )
+  )(validation)
+}
+
+const mapStateToProps = state => {
+  return { survey: SurveyState.getSurvey(state) }
```

Comment:
I wouldn't pass the entire survey to the component

Reply:
yes... me neither.. but how can we do it?